cura/Settings/MachineManager.py

```diff
@@ -1631,33 +1631,49 @@ def setQualityGroup(self, quality_group: "QualityGroup", no_dialog: bool = False
     #        Examples:
     #          - "my_profile - Fine" (only based on a default quality, no intent involved)
     #          - "my_profile - Engineering - Fine" (based on an intent)
-    @pyqtProperty("QVariantMap", notify = activeQualityDisplayNameChanged)
-    def activeQualityDisplayNameMap(self) -> Dict[str, str]:
-        global_stack = self._application.getGlobalContainerStack()
-        if global_stack is None:
-            return {"main": "",
-                    "suffix": ""}
+    @pyqtProperty("QList<QString>", notify = activeQualityDisplayNameChanged)
+    def activeQualityDisplayNameStringParts(self) -> [str]:
```

Comment:
```suggestion
    def activeQualityDisplayNameStringParts(self) -> List[str]:
```

You'll need the import too

Reply:
List is already imported 👍